### PR TITLE
[ticket/16015] Add new hook core.ucp_profile_avatar_upload_validation

### PR DESCRIPTION
**Pull request by xchwarze**
Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**): PHPBB3-16015

https://tracker.phpbb.com/browse/PHPBB3-16015

### CODE REVIEW

**Comment by 3D-I** [comment]
Should amend the commit message, like:

`[ticket/16015] Add new hook core.ucp_profile_avatar_upload_validation`

**Comment by xchwarze** [comment]
> 
> 
> Should amend the commit message, like:
> 
> `[ticket/16015] Add new hook core.ucp_profile_avatar_upload_validation`

Done!

**Comment by 3D-I** [comment]
Well, that shoud follow the https://area51.phpbb.com/docs/dev/3.2.x/development/git.html#commit-messages docs.

```
[ticket/16015] Add new hook core.ucp_profile_avatar_upload_validation

PHPBB3-16015
```

File: phpBB/phpbb/avatar/driver/remote.php [hunk 1]
```diff
@@ -84,6 +84,24 @@ public function process_form($request, $template, $user, $row, &$error)
...
+		 * @var	string	$url	Image url
+		 * @var	string	$width	Image width
+		 * @var	string	$height	Image height
+		 * @var	array	$error	Error message array
+		 * @since 3.2.6-RC1
+		 */
+		$vars = array('url', 'width', 'height', 'error');
+		extract($phpbb_dispatcher->trigger_event('core.ucp_profile_avatar_upload_validation', compact($vars)));
+
+		if (!empty($error))
```
Comment (3D-I) [inline phpBB/phpbb/avatar/driver/remote.php]: It looks redundant this block, I think you should now remove the lines

https://github.com/phpbb/phpbb/blob/fe09a8a4b3c75babdbd21729491a35bd7f861a94/phpBB/phpbb/avatar/driver/remote.php#L82-L85

Comment (xchwarze) [reply phpBB/phpbb/avatar/driver/remote.php]: this block is for show error from custom validation

Comment (3D-I) [reply phpBB/phpbb/avatar/driver/remote.php]: The errors from custom validation should be added to the existing array.

**Comment by 3D-I** [comment]
And missing tests.

File: phpBB/phpbb/avatar/driver/remote.php (hunk 1 above)
Comment (xchwarze) [reply phpBB/phpbb/avatar/driver/remote.php]: they are added inside the hook.

example:

`
class listener implements EventSubscriberInterface 
{

	static public function getSubscribedEvents()
	{
		return array(
			'core.ucp_profile_avatar_upload_validation'     => 'avatar_upload_validation',
		);
	}

    public function avatar_upload_validation($event)
    {
        if ($event['url'] === 'http://test.com/image.jpg')
        {
            $event['error'] = array_merge($event['error'], array('test.com is not a valid image hosting!'));
        }
    }
`

Like the rest of the validators of the file when it finds an error the function returns false

Comment (3D-I) [reply phpBB/phpbb/avatar/driver/remote.php]:
```
			$error = $event['error'];
			if ($event['url'] === 'http://test.com/image.jpg')
			{
				$error[] = $this->language->lang('NOT_A_VALID_IMG_HOSTING');
			}
			$event['error'] = $error;
```
or use the update_subarray() method.

Comment (3D-I) [reply phpBB/phpbb/avatar/driver/remote.php]: Also, `$phpbb_dispatcher` is not available https://travis-ci.org/phpbb/phpbb/jobs/520054536#L1357

Comment (xchwarze) [reply phpBB/phpbb/avatar/driver/remote.php]: fixed

**Comment by xchwarze** [comment]
Travis fails in  PHP: 5.4 DB=sqlite3 job, but seems to by a build error.

with same code https://travis-ci.org/phpbb/phpbb/jobs/520117846 its ok

**Comment by marc1706** [comment]
We have deprecated the remote avatar feature and will remove it soon. Are you sure you still need this event?

**Comment by xchwarze** [comment]
> 
> 
> We have deprecated the remote avatar feature and will remove it soon. Are you sure you still need this event?

mmm i use this event in my board

**Pull request merged**